bluesky/suspenders.py

```diff
@@ -27,37 +22,44 @@ class PVSuspenderBase(metaclass=ABCMeta):
         The event loop to work on
 
     """
-    def __init__(self, RE, pv_name, *, sleep=0, loop=None):
+    def __init__(self, signal, *, sleep=0, loop=None):
         """
         """
         if loop is None:
             loop = asyncio.get_event_loop()
-        self._loop = loop
-        self.RE = RE
+        self._loop = None
+        self.RE = None
         self._ev = None
+        self._alarm = False
         self._sleep = sleep
         self._lock = Lock()
-        self._pv = epics.PV(pv_name, auto_monitor=True)
-        self._indx = None
-        self.install()
+        self._sig = signal
 
-    def install(self):
+    def install(self, RE, *, event_type=None):
         '''Install callback on PV
 
         This (re)installs the required callbacks at the pyepics level
+
+        Parameters
+        ----------
+
+        RE : RunEngine
+            The run engine instance this should work on
+
+        event_type : str, optional
+            The event type (subscription type) to watch
         '''
-        if self._indx is not None:
-            self.remove()
-        self._indx = self._pv.add_callback(self)
+        self.RE = RE
+        self._loop = RE.loop
```

Comment:
Below, you get `loop` by accessing `self.RE.loop`. I think you can remove this private attribute entirely. I can't think of a reason that `self._loop` should be anything but the loop that `self.RE` is using.